starport/services/network/join.go

```diff
@@ -0,0 +1,168 @@
+package network
+
+import (
+	"context"
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	launchtypes "github.com/tendermint/spn/x/launch/types"
+	"github.com/tendermint/starport/starport/pkg/events"
+)
+
+// Join creates the RequestAddValidator message into the SPN
+func (b *Builder) Join(
+	validatorMsg,
+	accountMsg sdk.Msg,
+) (string, error) {
+	msgs := make([]sdk.Msg, 0)
+	if accountMsg != nil {
+		msgs = append(msgs, accountMsg)
+	}
+	if validatorMsg != nil {
+		msgs = append(msgs, validatorMsg)
+	}
+	b.ev.Send(events.New(events.StatusOngoing, "Broadcasting transactions"))
+	response, err := b.cosmos.BroadcastTx(b.account.Name, msgs...)
+	if err != nil {
+		return "", err
+	}
+
+	out, err := b.cosmos.Context.Codec.MarshalJSON(response)
+	if err != nil {
+		return "", err
+	}
+	b.ev.Send(events.New(events.StatusDone, "Transactions broadcasted"))
+
+	return string(out), err
+}
+
+// CreateValidatorRequestMsg creates an add AddValidator request message
+func (b *Builder) CreateValidatorRequestMsg(
+	ctx context.Context,
+	launchID uint64,
+	peer,
+	valAddress string,
+	gentx,
+	consPubKey []byte,
+	selfDelegation sdk.Coin,
+) (sdk.Msg, error) {
+	// Check if the validator request already exist
+	exist, launchID, err := b.checkRequestValidator(ctx, launchID, valAddress)
+	if err != nil {
+		return nil, err
+	}
+	if exist {
+		return nil, errors.New("validator already exist: " + valAddress)
+	}
+
+	return launchtypes.NewMsgRequestAddValidator(
+		valAddress,
+		launchID,
+		gentx,
+		consPubKey,
+		selfDelegation,
+		peer,
+	), nil
+}
+
+// CreateAccountRequestMsg creates an add AddAccount request message
+func (b *Builder) CreateAccountRequestMsg(
+	ctx context.Context,
+	chainHome string,
```

Comment:
If `--gentx` option is provided, the user may not have a initialized home for the chain. I think we should skip genesis verification in this case